spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryEnhancerFactory.java

```diff
@@ -17,29 +17,29 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.springframework.core.SpringProperties;
 import org.springframework.data.jpa.provider.PersistenceProvider;
+import org.springframework.lang.Nullable;
 import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
 
 /**
  * Encapsulates different strategies for the creation of a {@link QueryEnhancer} from a {@link DeclaredQuery}.
  *
  * @author Diego Krupitza
  * @author Greg Turnquist
  * @author Mark Paluch
+ * @author Christoph Strobl
  * @since 2.7.0
  */
 public final class QueryEnhancerFactory {
 
 	private static final Log LOG = LogFactory.getLog(QueryEnhancerFactory.class);
-
-	private static final boolean jSqlParserPresent = ClassUtils.isPresent("net.sf.jsqlparser.parser.JSqlParser",
-			QueryEnhancerFactory.class.getClassLoader());
+	private static final NativeQueryEnhancer NATIVE_QUERY_ENHANCER;
```

Comment:
With `SpringProperties` falling back to System Properties, wouldn't we want to keep the selection dynamic to allow changing system property values at runtime?

More with the rationale of setting the value later and not so much for flipping it.

Reply:
While we could do it, I'd rather not have this configurable. Allowing to change values at runtime might lead to interesting side effects in its best case, and we'd have to do the lookup every single time.